client/site-profiler/components/site-profiler-v2.tsx

```diff
@@ -105,21 +104,25 @@ export default function SiteProfilerV2( props: Props ) {
 		);
 	}
 
-	const showGetReportForm = !! showBasicMetrics && !! url && isGetReportFormOpen;
+	const showGetReportForm = !! url && isGetReportFormOpen;
 
-	const { data: performanceMetrics } = useUrlPerformanceMetricsQuery(
-		basicMetrics?.final_url,
-		basicMetrics?.token
-	);
+	const { data: performanceMetrics } = useUrlPerformanceMetricsQuery( routerDomain, hash );
 
-	const performanceCategory = getPerformanceCategory( performanceMetrics );
+	const { final_url: finalUrl, token } = basicMetrics || {};
+
+	useEffect( () => {
+		if ( finalUrl && token ) {
+			page( `/site-profiler/report/${ token }/${ finalUrl }` );
```

Comment:
Having the `.....wh8s/https://example.com` HTTP scheme in the URL doesn't look good, IMHO. I have made a few changes to backend validation to allow the domain as input instead of needing the full URL. After that change, we can pass the domain instead of final URL.

Reply:
>Having the .....wh8s/https://example.com HTTP scheme in the URL doesn't look good

Can you expand on your concerns? That is working like that in the current version in production
![CleanShot 2024-06-06 at 10 22 29@2x](https://github.com/Automattic/wp-calypso/assets/3519124/8ed82110-9230-4226-99db-2fc7d76194e8)